frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedColumns.jsx

```diff
@@ -28,7 +28,9 @@ export default class ChartSettingOrderedColumns extends Component {
   };
 
   handleSortEnd = ({ oldIndex, newIndex }) => {
-    const fields = [...this.props.value];
+    const fields = [...this.props.value].sort((a, b) =>
+      a.enabled && !b.enabled ? -1 : 1,
+    );
```

Comment:
@npfitz and @alxnddr , I would appreciate your eyes on this line. This was necessary because column reordering didn't work properly for object detail viz because the `value` prop here contained all columns, even disabled columns. This behaves differently for table viz, because we re-run the query every time we show or hide a column (which is probably not ideal), and that removes them from this value array. 

![Screen Shot 2023-01-20 at 2 39 53 PM](https://user-images.githubusercontent.com/30528226/213810035-d5917264-b47e-4ca1-a1ea-51655e163c3a.png)


The behavior i was seeing was that the enabled columns list would have, for instances, indexes `0,6,7` when columns 1-5 were hidden, but the ordering component would re-index those as `0,1,2`. So if you wanted to move the first item to the end of the list, it would move from position 0 -> 2 ... which would still order at the beginning because the objectDetail component still things there are 8 columns.

By making sure that `enabled` columns are always first, we ensure that we're properly sorting the order only amongst enabled columns and we get the intended behavior. I've also tested it on the table component, and it seems to work properly there as well. It also seems to work properly with pivot tables.

Nonetheless, I'd appreciate you throwing any red flags you see here, or maybe suggesting an alternative approach if you can think of one.

Reply:
Uh oh, one red flag I can see is that I'm removing that settings component in my PR (though I don't necessarily need to). Maybe we can sync up and see if there are alternatives

Reply:
This has now been updated to use the new `ChartSettingsColumnEditor` and `ChartSettingOrderedSimple` components

Reply:
Woo!